server/src/main/java/org/elasticsearch/indices/recovery/RecoveryState.java

```diff
@@ -451,13 +467,23 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         }
     }
 
-    public static class Translog extends Timer implements ToXContentFragment, Streamable {
+    public static class Translog extends Timer implements ToXContentFragment, Streamable, Writeable {
```

Comment:
I think we can remove `Streamable` from `Translog` then remove `readFrom` entirely.